documentation/specs/static-graph.md

```diff
@@ -227,34 +140,107 @@ We'll represent the project reference protocols as items in MSBuild. The target
+
+### Single project isolated builds
+When building a single project in isolation, all project references' build results must be provided to the project externally. Specifically, the results will need to be [deserialized](#deserialization) from files and loaded into the build result cache in memory.
+
+Because of this, single project isolated builds is quite restrictive and is not intended to be used directly by end-users. Instead the scenario is intended for higher-order build engines which support caching and [distribution](#distribution).
+
+There is also the possibility for these higher-order build engines and even Visual Studio to enable extremely fast incremental builds for a project. For example, when all project references' build results are provided (and validated as up to date by that higher-order build engine), there is no need to evaluate or execute any targets on any other project.
+
+These incremental builds can even be extended to multiple projects by keeping a project graph in memory as well as the last build result for each node and whether that build result is valid. The higher-order build engine can then itself traverse the graph and do single project isolated builds for projects which are not currently up to date.
 
 ## Distribution
-To support distribution with export targets (eg. for the MS internal build engine), we need a solution for a project and a dependency building on different machines.
-- Option 1 (easier): Run export targets only for dependencies, no build.
-  - Requires re-evaluation of the dependency, but no big MSBuild support beyond allowing the no-Build mode for dependencies.
-- Option 2 (better): Serialize the export target state
-  - Best for performance, but requires a serialization protocol and a way to specify how to import the state
-  - Possibly same mechanism as the persistent cross-build cache described above.
+To support distribution for isolated builds (eg. for the MS internal build engine), we need a solution for a project and a dependency building on different machines. To facilitate this, we need to serialize the build result cache to disk on the source machine which built the project reference, and deserialize it on the destination machine which is about to build the project referencing the project reference.
+
+In fact, this same mechanism can be used for non-distributed builds as well, for example for incremental builds. The caller would still need to ensure that the build result cache was valid (ie. the project reference project was not changed since the build result cache for it was produced), but the benefit would be that only the current project being built would need to be evaluated and only targets on that project would need to execute.
+
+### Serialization
```

Comment:
I'll be working on the Serialization, Deserialization, and I/O tracking sections this week. I wanted to get this update out of the way since it doesn't involve any behavioral changes to the spec, just some reorganization and clarification.